addons/web/views/webclient_templates.xml

```diff
@@ -463,6 +463,12 @@
         <xpath expr="//head/meta[last()]" position="after">
             <meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no"/>
         </xpath>
+        <xpath expr="//head" position="inside">
+            <script type="text/javascript">
+                const userTZ = Intl.DateTimeFormat().resolvedOptions().timeZone;
+                document.cookie = `tz=${userTZ}; path=/`;
+            </script>
```

Comment:
@aab-odoo I had not paid attention to this before, but is there any reason not to merge this `script` block with the [one just below](https://github.com/odoo-dev/odoo/blob/c985c7834088d738e09bbf89873e93b4781d8a62/addons/web/views/webclient_templates.xml#L477-L479) that sets `odoo.session_info`?

Additionally, I don't know if we can measure the cost of assigning `document.cookie` on every page, but @antonylesuisse suggested a quick test based on `document.cookie.indexOf('; tz=')` could save the effort of re-setting the cookie when it is already set, without having to parse the whole cookie string. I have no idea if that would be better on average, or if we should just rely on the browsers having efficient cookie jar implementations.

Reply:
@odony There is no particular reason to have another script for this. I guess I did this at first for testing, and never think of changing it afterwards. Do you want me to merge both scripts?

For the other remark, I would keep it as it is. I can't find any hint on the internet about poor performances of setting a cookie, and it is simpler this way.

Reply:
@aab-odoo yes please, it would be great to merge them. I'm not too sure how useful the cookie optimization is, but I don't think it can hurt anything, and it's just an extra line, no? @antonylesuisse seemed to want it, but you could probably convince him ;-)

Another noteworthy point: I've noticed that this new mechanism overlaps with what has been done in `website` at 17e84025239ed78716ed5d5d92c0ee0e10fbfb29 by @awa-odoo and @dbeguin. Their version did cause extra requests for visitors, which we wanted to avoid. Looks like we could merge the two mechanisms: have website_visitors use the cookie info when it's available, and get rid of the extra request and localStorage data. That should be considered a fix for saas-13.5.

Reply:
@odony Good suggestion. Thanks! Task created in our pipe : https://www.odoo.com/web?#id=2333825&action=333&active_id=965&model=project.task&view_type=form&cids=1&menu_id=4720

Reply:
@odony Test added and scripts merged in PR https://github.com/odoo/odoo/pull/57147